### PR TITLE
Extend Object prototype for message instances

### PR DESCRIPTION
**Pull request by erichiggins0**
After https://github.com/timostamm/protobuf-ts/pull/582, message instances don't behave exactly like simple objects, as their prototype only includes a `Symbol("protobuf-ts/message-type")` field and is missing fields normally present in a simple object (e.g. `constructor`). For us, this specifically causes an issue when using [mobx](https://github.com/mobxjs/mobx)  to observe changes to message instances - it only allows observing objects that have a `constructor` method in their prototype that matches the method for simple objects.

This fixes the problem by copying the default object prototype and adding a `Symbol("protobuf-ts/message-type")` field, rather than starting from an empty prototype.

### CODE REVIEW

File: packages/runtime/src/message-type.ts [hunk 1]
```diff
@@ -67,7 +67,14 @@ export class MessageType<T extends object> implements IMessageType<T> {
...
         this.options = options ?? {};
-        this.messagePrototype = Object.defineProperty({}, MESSAGE_TYPE, { value: this });
+        this.messagePrototype = this.messagePrototype = Object.defineProperty(
+            Object.create(
+              null,
+              Object.getOwnPropertyDescriptors(Object.getPrototypeOf({}))
+            ),
+            MESSAGE_TYPE,
+            { value: this },
+        );
```
Comment (jcready) [inline packages/runtime/src/message-type.ts:77]: We should probably store a reference to the object property descriptors just once outside the class:
```ts
const baseDescriptors = Object.getOwnPropertyDescriptors(Object.getPrototypeOf({}));
```

We can also simplify this a bit
```suggestion
        this.messagePrototype = Object.create(null, { ...baseDescriptors, [MESSAGE_TYPE]: { value: this } });
```

Comment (erichiggins0) [reply packages/runtime/src/message-type.ts]: @jcready done!

**Comment by timostamm** [comment]
@erichiggins0, CI was failing because of an issue with go. Pushed up https://github.com/timostamm/protobuf-ts/pull/636 to your fork.

**Comment by timostamm** [comment]
Released in [v2.9.4](https://github.com/timostamm/protobuf-ts/releases/tag/v2.9.4)

**Pull request merged**